buildchow/templates/product-template.yaml

```diff
@@ -117,39 +109,44 @@
+          maven-version: maven-3.2.5
           private-repository: false
           properties:
-            - artifact=org.jboss.tools.releng:jbosstools-releng-publish:4.3.0.Alpha1-SNAPSHOT:zip
+            - artifact=org.jboss.tools.releng:jbosstools-releng-publish:4.3.0.CR1-SNAPSHOT:zip
             - outputDirectory=${{WORKSPACE}}/sources
             - trimVersion=true
             - mdep.stripClassifier=true
             - mdep.stripVersion=true
       - shell: |
-               # move the non-pack200 site zip into the temporary folder
-               mv ${{WORKSPACE}}/sources/product/site/target/site_assembly.zip* ${{WORKSPACE}}/tmp-target/
+               # replace the update-core zip + sha256 with the non-pack200 versions
+               mv -f ${{WORKSPACE}}/sources/product/site/target/com.jboss.devstudio.core.site-*.zip* ${{WORKSPACE}}/tmp-target/
                # purge the target folder built w/o pack200
                rm -fr ${{WORKSPACE}}/sources/product/site/target
                # move the temporary folder back to where it should be
                mv ${{WORKSPACE}}/tmp-target ${{WORKSPACE}}/sources/product/site/target
-               
-               # replace the update-core zip + MD5 with the new one
-               cp -f ${{WORKSPACE}}/sources/product/site/target/site_assembly.zip ${{WORKSPACE}}/sources/product/results/target/jboss-devstudio-*updatesite-core.zip
-               z=`find ${{WORKSPACE}}/sources/product/results/target/ -name "jboss-devstudio-*updatesite-core.zip"`
-               for m in $(md5sum ${{z}}); do if [[ $m != ${{z}} ]]; then echo $m &gt; ${{z}}.MD5; fi; done
-               
-               cd ${{WORKSPACE}}/sources/
-               export skipRevisionCheckWhenPublishing=${{skipRevisionCheckWhenPublishing}}
-               DESTINATION=/qa/services/http/binaries/RHDS . publish/publish.sh
+
+               ###############################################
+
+               if [[ ${{skipRevisionCheckWhenPublishing}} == "true" ]] || [[ $(. ${{WORKSPACE}}/sources/util/checkLatestPublishedSHA.sh -s ${{WORKSPACE}}/sources/product/site/target/repository -t http://www.qa.jboss.com/binaries/RHDS/9.0/snapshots/builds/${{JOB_NAME}}/latest/all/repo/) == "true" ]]; then
+                 cd ${{WORKSPACE}}/sources/product/results
+                 M2_HOME=/qa/tools/opt/apache-maven-3.2.5/
+                 $M2_HOME/bin/mvn deploy -U -e -Pdeploy-to-jboss.org -Dmaven.repo.local=$WORKSPACE/.repository -Djbosstools_site_stream=${{jbosstools_site_stream}} ${{MAVEN_FLAGS}}
+               else
+                 echo "Publish cancelled (nothing to do). Skip this check with skipRevisionCheckWhenPublishing=true"
+                 BUILD_DESCRIPTION="NOT PUBLISHED: UNCHANGED"
+               fi
+
     publishers:
         - description-setter: 
             regexp: "BUILD_DESCRIPTION='(.+)'"
             description: "\\1"
-            ## TODO: downstreamtrigger
+        - raw:
+            xml: |
```

Comment:
any reason to not use the native built-in support ? 

http://docs.openstack.org/infra/jenkins-job-builder/publishers.html#publishers.downstream-ext